File: TwoSum2.cs
public class Solution 
{
    public int[] TwoSum(int[] nums, int target) 
    {
        var dict = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if(dict.ContainsKey(target-nums[i]))// checks if compliment is in dict, i.e. 2+4=6,6 is target
            {
                return new int[]{dict[target-nums[i]], i};
            }
            else if(!dict.ContainsKey(nums[i]))// handles duplicates in array
            {
                dict.Add(nums[i], i);
            }
        }
        return null;  
    }
}

File: twoSum2.java
class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        int[] dummy = nums.clone();
        int N = nums.length;
        Arrays.sort(nums); // sort nums so that numbers increase from 
                          // 'right' to 'left' to be used below
        int left = 0; int right = N - 1;

        while (nums[left]+nums[right]!=target)
        {
            if(nums[left]+nums[right]>target)
            {
                right--;
            }
            else
            {
                left++;
            }
        }
        int[] result = new int[2];
        for(int i = 0; i < nums.length; i++)
        {
            if(dummy[i] == nums[left])
            {
                result[0] = i;
            }
        }

        for(int j = N -1; j >= 0; j--)
        {
            if(dummy[j] == nums[right])
            {
                result[1] = j;
            }
        }

        return result;
    }
}
